whois-commons/src/main/java/net/ripe/db/whois/common/rpsl/DummifierRC.java

```diff
@@ -1,29 +1,99 @@
 
+import com.google.common.base.Splitter;
 import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+import net.ripe.db.whois.common.rpsl.transform.FilterChangedFunction;
+import org.apache.commons.lang.Validate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.stereotype.Component;
 
+import java.util.Iterator;
 import java.util.List;
+import java.util.Set;
+
+import static net.ripe.db.whois.common.rpsl.AttributeType.ABUSE_MAILBOX;
+import static net.ripe.db.whois.common.rpsl.AttributeType.ADDRESS;
+import static net.ripe.db.whois.common.rpsl.AttributeType.AUTH;
+import static net.ripe.db.whois.common.rpsl.AttributeType.E_MAIL;
+import static net.ripe.db.whois.common.rpsl.AttributeType.FAX_NO;
+import static net.ripe.db.whois.common.rpsl.AttributeType.IRT_NFY;
+import static net.ripe.db.whois.common.rpsl.AttributeType.MNT_NFY;
+import static net.ripe.db.whois.common.rpsl.AttributeType.NOTIFY;
+import static net.ripe.db.whois.common.rpsl.AttributeType.PERSON;
+import static net.ripe.db.whois.common.rpsl.AttributeType.PHONE;
+import static net.ripe.db.whois.common.rpsl.AttributeType.REF_NFY;
+import static net.ripe.db.whois.common.rpsl.AttributeType.UPD_TO;
 
 @Component
-public class DummifierRC extends DummifierCurrent {
+public class DummifierRC implements Dummifier {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DummifierRC.class);
+
+    private static final String PERSON_REPLACEMENT = "Name Removed";
+    private static final String FILTERED_APPENDIX = " # Filtered";
+    private static final Splitter EMAIL_SPLITTER = Splitter.on('@');
+    private static final Splitter SPACE_SPLITTER = Splitter.on(' ');
+
+    private static final Set<AttributeType> EMAIL_ATTRIBUTES = Sets.immutableEnumSet(E_MAIL, NOTIFY, REF_NFY, IRT_NFY, MNT_NFY, UPD_TO);
+    private static final Set<AttributeType> PHONE_FAX_ATTRIBUTES = Sets.immutableEnumSet(PHONE, FAX_NO);
+    private static final FilterChangedFunction FILTER_CHANGED_FUNCTION = new FilterChangedFunction();
 
     @Override
     public RpslObject dummify(final int version, final RpslObject rpslObject) {
+        final ObjectType objectType = rpslObject.getType();
+        Validate.isTrue(isAllowed(version, rpslObject), "The version is not supported by this dummifier", version);
+
         final List<RpslAttribute> attributes = Lists.newArrayList(rpslObject.getAttributes());
 
+        RpslAttribute lastAddressLine = null;
+        int lastAddressLineIndex = 0;
+
+
         for (int i = 0; i < attributes.size(); i++) {
             RpslAttribute replacement = attributes.get(i);
 
-            replacement = dummifyOrgName(rpslObject, replacement);
-            replacement = dummifyDescr(replacement);
-            replacement = dummifyRemarks(replacement);
-            replacement = dummifyOwner(replacement);
+            replacement = dummifyAttributes(rpslObject, attributes, i, replacement);
+
```

Comment:
What does i signifies? Can you not pass the counter of for loop in a method. It is not so readable

Reply:
The original code seems more readable for me

Reply:
Done